### PR TITLE
Add Emacs packages: apheleia, isearch-prop, isearch-plus

### PR DESCRIPTION
**Pull request by leungbk**
###### Motivation for this change

Adding Emacs packages not in Melpa

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by leungbk** [comment]
Ping @adisbladis

**Comment by pbogdan** [comment]
I have not reviewed or tested this PR yet but would very much like to see `apheleia` included so :+1: in principle, thank you for the PR (nudges @adisbladis gently).

**Comment by leungbk** [comment]
When I made this PR initially, I hadn't realized that my info needed to be in the maintainers file. I added my info in https://github.com/NixOS/nixpkgs/pull/88005. Not sure if I should add it here instead and modify that other PR.

**Comment by adisbladis** [comment]
As a general rule I'm reluctant to add manually package emacs packages, they tend to not get updated for many, many years and get broken over time.

I'd much rather pester the respective upstreams for a Melpa push if possible.

**Comment by leungbk** [comment]
> As a general rule I'm reluctant to add manually package emacs packages, they tend to not get updated for many, many years and get broken over time.
> 
> I'd much rather pester the respective upstreams for a Melpa push if possible.

I've asked for Melpa recipes from the package managers of: apheleia, evil-markdown, helm-linux-disks, helm-wikipedia, helm-taskrunner, mu4e-patch, and org-auto-expand. 

The maintainer of git-undo has expressed interest in merging it into core Emacs. The maintainer of this fork of evil-numbers has gotten in contact in the past with the inactive maintainer of the upstream evil-numbers, but they seem to have broken off contact.

The maintainers of youtube-dl, isearch-plus, and isearch-prop don't want to publish to Melpa.

File: pkgs/applications/editors/emacs-modes/apheleia/default.nix [hunk 1]
```diff
@@ -0,0 +1,31 @@
+{ stdenv, fetchFromGitHub, emacs }:
+
+stdenv.mkDerivation {
+  name = "apheleia";
+
+  src = fetchFromGitHub {
+    owner = "raxod502";
+    repo = "apheleia";
+    rev = "3e342632b834a78f31ead48b94392e00dba1542c";
```
Comment (SuperSandro2000) [inline pkgs/applications/editors/emacs-modes/apheleia/default.nix:9]: Please track a tag.

File: pkgs/applications/editors/emacs-modes/apheleia/default.nix [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+  '';
+
+  installPhase = ''
+    install -d $out/share/emacs/site-lisp
+    install *.el *.elc $out/share/emacs/site-lisp
+  '';
+
+  meta = {
+    description = "Reformat buffer stably";
+    homepage = https://github.com/raxod502/apheleia;
```
Comment (SuperSandro2000) [inline pkgs/applications/editors/emacs-modes/apheleia/default.nix:26]:
```suggestion
    homepage = "https://github.com/raxod502/apheleia";
```

File: pkgs/applications/editors/emacs-modes/evil-markdown/default.nix [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+  '';
+
+  installPhase = ''
+    install -d $out/share/emacs/site-lisp
+    install *.el *.elc $out/share/emacs/site-lisp
+  '';
+
+  meta = {
+    description = "Vim-like keybindings for markdown-mode";
+    homepage = https://github.com/Somelauw/evil-markdown;
```
Comment (SuperSandro2000) [inline pkgs/applications/editors/emacs-modes/evil-markdown/default.nix:36]:
```suggestion
    homepage = "https://github.com/Somelauw/evil-markdown";
```

File: pkgs/applications/editors/emacs-modes/evil-numbers-janpath/default.nix [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+  '';
+
+  installPhase = ''
+    install -d $out/share/emacs/site-lisp
+    install *.el *.elc $out/share/emacs/site-lisp
+  '';
+
+  meta = {
+    description = "Increment and decrement numbers as in Vim";
+    homepage = https://github.com/janpath/evil-numbers;
```
Comment (SuperSandro2000) [inline pkgs/applications/editors/emacs-modes/evil-numbers-janpath/default.nix:35]:
```suggestion
    homepage = "https://github.com/janpath/evil-numbers";
```

File: pkgs/applications/editors/emacs-modes/git-undo/default.nix [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+  '';
+
+  installPhase = ''
+    install -d $out/share/emacs/site-lisp
+    install *.el *.elc $out/share/emacs/site-lisp
+  '';
+
+  meta = {
+    description = "Revert region to most recent Git-historical version";
+    homepage = https://github.com/jwiegley/git-undo-el;
```
Comment (SuperSandro2000) [inline pkgs/applications/editors/emacs-modes/git-undo/default.nix:26]:
```suggestion
    homepage = "https://github.com/jwiegley/git-undo-el";
```

File: pkgs/applications/editors/emacs-modes/helm-linux-disks/default.nix [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+  '';
+
+  installPhase = ''
+    install -d $out/share/emacs/site-lisp
+    install *.el *.elc $out/share/emacs/site-lisp
+  '';
+
+  meta = {
+    description = "Mount/unmount volumes in Linux with Helm";
+    homepage = https://github.com/akirak/helm-linux-disks;
```
Comment (SuperSandro2000) [inline pkgs/applications/editors/emacs-modes/helm-linux-disks/default.nix:35]:
```suggestion
    homepage = "https://github.com/akirak/helm-linux-disks";
```

**Review by SuperSandro2000 — COMMENTED** [review]
Can you also use 

```
meta = with stdenv.lib; {
```
?

**Comment by adisbladis** [comment]
@leungbk I've done a bunch of minor nit fixups, but I lost energy once I realised that `name = "foo"` was set (all packages in this PR lacks a version identifier).

Could you fix that and maybe make sure that the packages are up to date? Once that's done we can go ahead with this.

**Comment by AndersonTorres** [comment]
Can I get those for me? :3

File: pkgs/applications/editors/emacs-modes/apheleia/default.nix (hunk 1 above)
Comment (leungbk) [reply pkgs/applications/editors/emacs-modes/apheleia/default.nix]: The latest rev of Apheleia has a fix for a fairly annoying bug (https://github.com/raxod502/apheleia/commit/f865c16), so I would prefer to use this instead.

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

**Review by AndersonTorres — APPROVED** [review]

**Pull request merged**